source/Editor.js

```diff
@@ -1993,11 +1993,49 @@ var keyHandlers = {
             // If the commonAncestor was in a mathjax equation, delete the parent. Namely the mathjax equation.
             if (range.commonAncestorContainer && 
                 range.commonAncestorContainer.parentElement && 
-                checkInSVG.test(range.commonAncestorContainer.nodeName)
+                (checkInSVG.test(range.commonAncestorContainer.nodeName) || checkInSVG.test(range.commonAncestorContainer.parentElement.nodeName))
```

Comment:
In that case I was talking about a structure that looked like.
SPAN.mathjax
     -> SCRIPT
             -> TEXT NODE (The browser passes this one).

I don't think it's too offensive to confirm that the parent of what was passed wasn't an SVG element or a Script tag. Those things should be delete anyway shouldn't they?